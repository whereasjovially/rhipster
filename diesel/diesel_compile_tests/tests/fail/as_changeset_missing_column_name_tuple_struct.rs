#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
        name -> Text,
        hair_color -> Nullable<Text>,
    }
}

#[derive(AsChangeset)]
#[table_name = "users"]
struct User(i32, #[column_name = "name"] String, String);

fn main() {}
