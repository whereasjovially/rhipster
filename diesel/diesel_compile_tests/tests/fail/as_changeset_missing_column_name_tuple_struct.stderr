error: All fields of tuple structs must be annotated with `#[column_name]`
  --> $DIR/as_changeset_missing_column_name_tuple_struct.rs:14:13
   |
14 | struct User(i32, #[column_name = "name"] String, String);
   |             ^^^

error: All fields of tuple structs must be annotated with `#[column_name]`
  --> $DIR/as_changeset_missing_column_name_tuple_struct.rs:14:50
   |
14 | struct User(i32, #[column_name = "name"] String, String);
   |                                                  ^^^^^^

error[E0412]: cannot find type `unknown_column` in module `users`
  --> $DIR/as_changeset_missing_column_name_tuple_struct.rs:14:13
   |
14 | struct User(i32, #[column_name = "name"] String, String);
   |             ^^^ not found in `users`

error[E0412]: cannot find type `unknown_column` in module `users`
  --> $DIR/as_changeset_missing_column_name_tuple_struct.rs:14:50
   |
14 | struct User(i32, #[column_name = "name"] String, String);
   |                                                  ^^^^^^ not found in `users`

error[E0425]: cannot find value `unknown_column` in module `users`
  --> $DIR/as_changeset_missing_column_name_tuple_struct.rs:14:13
   |
14 | struct User(i32, #[column_name = "name"] String, String);
   |             ^^^ not found in `users`

error[E0425]: cannot find value `unknown_column` in module `users`
  --> $DIR/as_changeset_missing_column_name_tuple_struct.rs:14:50
   |
14 | struct User(i32, #[column_name = "name"] String, String);
   |                                                  ^^^^^^ not found in `users`
